File: 0222) Count Complete Tree Nodes.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private int ans = 0;
    public int countNodes(TreeNode root) {
        helper(root);
        return ans;
    }
    public void helper(TreeNode root)
    {
        if(root == null)
        {
            return;
        }
        if(root.left != null && root.right != null)
        {
            ans++;
        }
        if(root.left == null || root.right == null)
        {
            ans++;
        }
        helper(root.left);
        helper(root.right);
    }
}

File: 0222) Count Complete Tree Nodes.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int countNodes(TreeNode* root) {
        ios::sync_with_stdio(0);
        cin.tie(0);
        cout.tie(0);
        int ans = 0;
        solve(root, ans);
        return ans;
    }
    void solve(TreeNode *root, int &ans)
    {
        if(root == nullptr)
        {
            return;
        }
        if(root->left && root->right)
        {
            ans++;
        }
        if(!root->left || !root->right)
        {
            ans++;
        }
        solve(root->left, ans);
        solve(root->right, ans);
    }
};
